packages/vite/src/node/config.ts

```diff
@@ -185,6 +186,23 @@ export interface UserConfig {
    * @deprecated use `resolve.dedupe` instead
    */
   dedupe?: string[]
+  /**
+   * worker bundle option
+   */
+  worker?: {
+    /**
+     * worker output format
+     */
+    format?: ModuleFormat
```

Comment:
Check PR from @kepta https://github.com/vitejs/vite/pull/5403/files, they are also passing `preserveEntrySignatures: false,`. We should understand why that was needed

And if the format is ES, the worker should have `{ type: module }` here?
https://github.com/vitejs/vite/blob/1f945f62bf4a722c95a7b8f9c14c32a6f2be5c3f/packages/vite/src/node/plugins/worker.ts#L83

Maybe we should work on supporting format in a separate PR?

Reply:
so we also need to export the option to config `{type: "module"}`?

Reply:
I think we can use the format option, if it is `es` we should use type module, no?

Reply:
yes, thank you😀

Reply:
Yes, let's only add the options that we know will work here.